packages/core/src/utils/getFieldDefaultValue.ts

```diff
@@ -21,7 +21,14 @@ const getFieldDefaultValue = (
   schema: FormupYupSchema,
   options?: getSchemaFieldOptions,
 ) => {
-  const schemaField = getSchemaField(name, schema, options);
+  const defaultOptions = {
+    returnSubtype: true,
+  };
+
+  const schemaField = getSchemaField(name, schema, {
+    ...defaultOptions,
+    ...options,
```

Comment:
If options is null this would break

```suggestion
    ...(options || {}),
```